### PR TITLE
Correct Lead/Lag + Null Offset behavior in the doc

### LINKED ISSUE #20835: [Discussion] Lead/Lag function's behavior when the offset is NULL
```java
// Presto TestLeadFunction.java
assertWindowQuery("lead(orderkey, null, -1) OVER (PARTITION BY orderstatus ORDER BY orderkey)",
        resultBuilder(TEST_SESSION, INTEGER, VARCHAR, INTEGER)
                .row(3, "F", null)
                .row(5, "F", null)
                .row(6, "F", null)
                .row(33, "F", null)
                .row(1, "O", null)
                .row(2, "O", null)
                .row(4, "O", null)
                .row(7, "O", null)
                .row(32, "O", null)
                .row(34, "O", null)
                .build());
```

In this test case, the offset is null, default value is specified as `-1`, but the expected value is `null` rather than `-1`, while in the doc([1]), it says that default value should be returned:

> If the offset is null or larger than the window, the default_value is returned, or if it is not specified null is returned.

`Lag` function has similar issue.

[1]. https://prestodb.io/docs/current/functions/window.html

### PR DESCRIPTION
**Pull request by xumingming**
## Description

Describe Lead/Lag + Null Offset behavior more clearly: when the offset is null, return null(not default value).
Fixes #20835

## Motivation and Context
N/A

## Impact
N/A

## Test Plan
N/A

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-docs/src/main/sphinx/functions/window.rst [hunk 1]
```diff
@@ -190,16 +190,18 @@ null for all rows, the ``default_value`` is returned, or if it is not specified,
 
 .. function:: lead(x[, offset [, default_value]]) -> [same as input]
 
-    Returns the value at ``offset`` rows after the current row in the window.
+    Returns the value at ``offset`` rows after the current row in the window partition.
     Offsets start at ``0``, which is the current row. The
     offset can be any scalar expression. The default ``offset`` is ``1``. If the
-    offset is null or larger than the window, the ``default_value`` is returned,
-    or if it is not specified ``null`` is returned.
+    offset is null, null is returned. If the offset refers to a row that is not
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/functions/window.rst:196]:
```suggestion
    offset is ``null``, ``null`` is returned. If the offset refers to a row that is not
```
Suggest formatting change for consistency with lines 198 and 207.

File: presto-docs/src/main/sphinx/functions/window.rst [hunk 2]
```diff
@@ -190,16 +190,18 @@ null for all rows, the ``default_value`` is returned, or if it is not specified,
...
 
 .. function:: lag(x[, offset [, default_value]]) -> [same as input]
 
-    Returns the value at ``offset`` rows before the current row in the window
+    Returns the value at ``offset`` rows before the current row in the window partition.
     Offsets start at ``0``, which is the current row. The
     offset can be any scalar expression. The default ``offset`` is ``1``. If the
-    offset is null or larger than the window, the ``default_value`` is returned,
-    or if it is not specified ``null`` is returned.
+    offset is null, null is returned. If the offset refers to a row that is not
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/functions/window.rst:205]:
```suggestion
    offset is null, null is returned. If the offset refers to a row that is not
```
Suggest formatting change for consistency with lines 198 and 207.

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the documentation! I have made a couple of formatting suggestions for consistency, but I noticed no other concerns.

**Comment by xumingming** [comment]
@steveburnett Addressed the review comments.

**Review by steveburnett — APPROVED** [review]
LGTM! (docs) 

Thanks for the doc!

**Comment by aditi-pandit** [comment]
Thanks @xumingming.

**Comment by mbasmanova** [comment]
@xumingming Folks, next time please check commit messages. We should not have "Address review comments" commits.

**Comment by xumingming** [comment]
@mbasmanova Ahh, I was thing leaving the commit which address the review comments separately would make the review more convenient to see what's changed incrementally. You prefer all commits rebased into one commit or separate commits with better commit messages?

**Comment by mbasmanova** [comment]
@xumingming See https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#pull-request-checklist

**Comment by mbasmanova** [comment]
@xumingming Presto project setup is different from Velox. Presto allows multi-commit PRs while Velox does not. When Presto PR is merged, all commits in the PR are rebased and merged as is. Here is what currently git log looks like:

```
2a18649d81 Add null stats for join probe
54edca6cf1 Address review comments
69bed7f13e Describe Lead/Lag + Null Offset behavior more clearly
b67703aece Support indicating partition transform when add column to Iceberg table
...
```

Notice the "Address review comments". This commit is not helpful and he commit "Describe Lead/Lag + Null Offset behavior more clearly" is not complete making it difficult to use gi blame / git log to research how the changes came to being.

When Velox PR is merged, all commits are squashed and a new commit message is created using PR title and description. Hence, in Velox, every PR ends up in a single commit on main. Hence, in Velox, it doesn't matter how many commits you have or what their commit messages. What matters is the PR title and description.

Hope this clarifies things. Let me know if you have any follow-up questions.

**Comment by xumingming** [comment]
@mbasmanova Thanks for the detailed explanation, got it now. I was indeed feel a bit confused at first, because I see several times "Address review comments" commit in Velox PRs. All things clear now.

**Pull request merged**